modules/custom/d_p_carousel/d_p_carousel.module

```diff
@@ -102,14 +104,56 @@ function d_p_carousel_preprocess_paragraph(&$variables)
        */
       $link = $paragraph->get('field_d_cta_link')->first();
       if (!is_null($link)) {
-        $variables['has_link'] = true;
+        $variables['has_link'] = TRUE;
         $variables['attributes']['href'][] = $link->getUrl()->toString();
       }
       else {
-        $variables['has_link'] = false;
+        $variables['has_link'] = FALSE;
       }
       break;
   }
 }
 
+/**
+ * Implements hook_form_alter().
+ */
+function d_p_carousel_form_alter(&$form, $form_state, $form_id) {
+  switch ($form_id) {
+    case 'node_content_page_form':
+      $widget = $form['field_page_section']['widget'];
+      $delta = range(0, $widget['#max_delta']);
+      if (!empty($widget)) {
+        foreach ($delta as $del) {
+          if (isset($widget[$del]['subform']['field_d_p_cs_item_reference'])) {
+            $form['field_page_section']['widget'][$del]['subform']['field_d_p_cs_item_reference']['widget'][0]['subform']['field_d_image']['widget'][0]['#cardinality'] = 1;
```

Comment:
['field_d_p_cs_item_reference']['widget'][0] - this will work only for first item